pkg/store/telemetry.go

```diff
@@ -86,3 +87,60 @@ type NoopSeriesStatsAggregator struct{}
 func (s *NoopSeriesStatsAggregator) Aggregate(_ storepb.SeriesStatsCounter) {}
 
 func (s *NoopSeriesStatsAggregator) Observe(_ float64) {}
+
+// instrumentedStoreServer is a storepb.StoreServer that exposes metrics about Series requests.
+type instrumentedStoreServer struct {
+	storepb.StoreServer
+	seriesRequested prometheus.Histogram
+	chunksRequested prometheus.Histogram
+}
+
+// NewInstrumentedStoreServer creates a new instrumentedStoreServer.
+func NewInstrumentedStoreServer(reg prometheus.Registerer, store storepb.StoreServer) storepb.StoreServer {
+	return &instrumentedStoreServer{
+		StoreServer: store,
+		seriesRequested: promauto.With(reg).NewHistogram(prometheus.HistogramOpts{
+			Name:    "thanos_store_server_series_requested",
+			Help:    "Number of requested series for Series calls",
+			Buckets: []float64{1, 10, 100, 1000, 10000, 100000},
```

Comment:
Maybe we can bump this even more? I have some servers sending millions of series in certain calls.

Reply:
With native histograms we won't have to configure buckets anymore. Let's add a couple more buckets for chunks temporarily until we have complete native histograms support.